draft-bormann-jsonpath-iregexp.md

```diff
@@ -134,7 +141,7 @@ Implementations which interpret I-Regexps MUST
 yield Boolean results as specified in {{XSD-2}}.
 (See also {{xsd-regexps}}.)
 
-Implementations MAY yield errors for regular expressions which are not I-Regexps.
+Strict implementations MAY yield errors for regular expressions which are not I-Regexps.
```

Comment:
Should we be stronger than MAY here?  What is the alternative action if a strict implementation receives an expression that's not iregexp?

Also later (line 241) you assert that strict implementations definitely raise errors.

Reply:
Not sure. I was trying to give plenty of leeway to implementations. A strict implementation _could_ choose to return `false` (i.e., no match) instead of returning an error. That wouldn't be my personal choice, but it might suit some.

If we can agree on this, I'll sort out the later anomaly on line 241.

Reply:
No, returning false is *not* acceptable for a checking ("strict") implementation.

Reply:
The point about "checking" is that non-conforming input is not silently accepted.
So things should be blowing up (during the syntax checking phase), not fail silently (during the matching phase).

Reply:
Erroring is good for interop and security. Failing silently is ok for security. Depends on your goals, but I'm not going to push this.